packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1081,6 +1089,9 @@ abstract class TextInputClient {
   /// Requests that this client perform the given action.
   void performAction(TextInputAction action);
 
+  /// Notify client about new content insertion from Android keyboard.
+  void commitContent(Map<String, dynamic> content);
```

Comment:
Would it be possible to split this out of TextInputClient, maybe onto its own method channel?  We're trying to keep TextInputClient slim to avoid breaking people that have implemented it.  I'm working on a better set up for this.

Reply:
Where is the preferred location?

Reply:
So there has been some discussion about this, we now have a [design doc](https://docs.google.com/document/d/1OxDsf_eot7TlsRn57z-1_dGbvuqnMxO7QEGIAmOCZtA/edit#).  That doc is still up for debate, so feel free to chime in if you have opinions.

If it makes sense for commitContent to be separate from TextInputClient, then I think I would recommend creating an entirely new method channel in system_channels.dart.  But does it make sense on its own, or is it inherently tied to TextInputClient?  Maybe it's not possible to use commitContent without opening an input connection and everything first via TextInputClient.

CC @tneotia